File: server/src/main/java/ftn/booking/model/Adventure.java
package ftn.booking.model;
import ftn.booking.model.enums.CancelationPrice;
import ftn.booking.model.enums.InstructorEquipment;
import ftn.booking.model.enums.RulesBehavior;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.lang.Nullable;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "adventures")
@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
public class Adventure {

    @Id
    @GeneratedValue( strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private Instructor instructor;

    private String name;

    private String address;

    private String description;

    private Integer maxNumOfPersons;


    @Enumerated
    @ElementCollection(targetClass = RulesBehavior.class)
    private List<RulesBehavior> rules = new ArrayList<>();

    @Enumerated
    @ElementCollection(targetClass = InstructorEquipment.class)
    private List<InstructorEquipment> equipment = new ArrayList<>();

    @Nullable
    private Integer roomNum;

    private Long price;

    @Nullable
    private Integer rate;

    private String additionalInfo;

    @Enumerated
    private CancelationPrice cancelationPrice;

}

File: client/src/app/dto/Adventure.ts
import { User } from "./user";

export class Adventure{
    id !: Number;
    instructorId !: Number;
    name !: String;
    address !: String;
    description !: String;
    maxNumOfPersons !: Number;
    rules!: String[];
    equipment !: String[];
    roomNum !: Number;
    price !: Number;
    rate !: Number;
    additionalInfo !: String;
    cancelationPrice !: String;
    numberOfActions !: Number;
    numberOfPastActions !: Number;
    instructorInfo !: User;
    hasSubscription !: Boolean;
   

}